### PR TITLE
ovl_Bg_Haka_Bombwall partially decompiled (1 NON_MATCHING)

### PR DESCRIPTION
**Pull request by sonicdcer**
https://decomp.me/scratch/DY4zR

func_80BD5E6C is identical to func_80BD4720 in ovl_Bg_Ikana_Bombwall (except for a few values), which is also NON_MATCHING feel free to try to match it, if you do, you'll be killing two birds with one stone.

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c [hunk 1]
```diff
@@ -28,50 +36,191 @@ const ActorInit Bg_Haka_Bombwall_InitVars = {
...
+        spBC.z = (Rand_ZeroOne() * 4.0f) - 2.0f;
+
+        Matrix_MultVec3f(&spC8, &spE0);
+        Matrix_MultVec3f(&spBC, &spD4);
+
+        spE0.x += this->dyna.actor.world.pos.x;
+        spE0.y += this->dyna.actor.world.pos.y;
+        spE0.z += this->dyna.actor.world.pos.z;
+
+        if ((i & 3) == 0) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c:119]: I guess i%4 doesn't match, right?
```suggestion
        if ((i % 4) == 0) {
```

File: src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c [hunk 2]
```diff
@@ -28,50 +36,191 @@ const ActorInit Bg_Haka_Bombwall_InitVars = {
...
+            temp_fs0 -= 150.0f;
+        }
+        spC8.x = temp_fs0;
+
+        phi_fs1 += 5;
+        spC8.y = phi_fs1;
+
+        spC8.z = (Rand_ZeroOne() * 20.0f) - 10.0f;
+
+        spBC.x = ((Rand_ZeroOne() - 0.5f) * 5.0f) + (temp_fs0 * 0.053333335f);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c:108]: Maybe `4.0f / 75.0f`?
```suggestion
        spBC.x = ((Rand_ZeroOne() - 0.5f) * 5.0f) + (temp_fs0 * (4.0f / 75.0f));
```

File: src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c [hunk 3]
```diff
@@ -28,50 +36,191 @@ const ActorInit Bg_Haka_Bombwall_InitVars = {
...
+        spE0.z += this->dyna.actor.world.pos.z;
+
+        if ((i & 3) == 0) {
+            phi_s0 = 32;
+            func_800BBFB0(play, &spE0, 60.0f, 2, 100, 120, 1);
+        } else {
+            phi_s0 = 64;
+        }
+
+        if (i & 1) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c:126]: Maybe?
```suggestion
        if (i % 2 != 0) {
```

File: src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c [hunk 4]
```diff
@@ -28,50 +36,191 @@ const ActorInit Bg_Haka_Bombwall_InitVars = {
...
+            phi_s0 |= 1;
+            phi_t0 = 1;
+        } else {
+            phi_t0 = 0;
+        }
+
+        if (D_80BD64CC[i & 3] >= 16) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c:133]:
```suggestion
        if (D_80BD64CC[i % ARRAY_COUNT(D_80BD64CC)] >= 16) {
```

File: src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.h [hunk 5]
```diff
@@ -3,15 +3,18 @@
...
 typedef struct BgHakaBombwall {
-    /* 0x000 */ Actor actor;
-    /* 0x144 */ char unk_144[0x64];
+    /* 0x000 */ DynaPolyActor dyna;
+    /* 0x15C */ ColliderCylinder collider;
     /* 0x1A8 */ BgHakaBombwallActionFunc actionFunc;
-    /* 0x1AC */ char unk_1AC[0x4];
+    /* 0x1AC */ s8 csTimer;
+    /* 0x1AD */ UNK_TYPE1 unk_1AC[0x3];
 } BgHakaBombwall; // size = 0x1B0
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.h:18]: Struct padding probably
```suggestion
    /* 0x1AC */ s8 csTimer;
} BgHakaBombwall; // size = 0x1B0
```

File: src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c (hunk 1 above)
Comment (sonicdcer) [reply src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c:119]: that function is currently NON_MATCHING, if you read the PR description, you'll find the corresponding scratch

File: src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c (hunk 4 above)
Comment (sonicdcer) [reply src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c:133]: Doesn't match. Again, please verify locally before doing PR reviews, the scratch is right there in the description

File: src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c [hunk 6]
```diff
@@ -28,50 +36,191 @@ const ActorInit Bg_Haka_Bombwall_InitVars = {
...
+            gravity = -550;
+        } else {
+            gravity = -450;
+        }
+
+        EffectSsKakera_Spawn(play, &spE0, &spD4, &spE0, gravity, phi_s0, 30, 0, 0, D_80BD64CC[i & 3], phi_t0, 0, 50, -1,
```
[the remark this answers is not in the document]
Comment (sonicdcer) [reply src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c:139]: doesn't match, please try it yourself before submitting a review

**Review by AngheloAlf — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**